### ISSUE TITLE
disallow $state(new SvelteSet) and similar

### ISSUE THREAD

**Issue by mustafa0x** [sveltejs/eslint-plugin-svelte#1052]
### Motivation

SvelteMap, SvelteSet, and SvelteURL don't need to be wrapped in $state. but that's not always obvious

### Description

per above

### Examples

```svelte
<!-- ✓ GOOD -->
<script>
const myset = new SvelteSet
</script>

<!-- ✗ BAD -->
<script>
const myset = $state(new SvelteSet)
</script>
```

**Comment by marekdedic** [comment]
Hmm, reacting quickly and not thoroughly enough, but this needs a little bit more nuance:

```ts

let myOptionalSet = $state(new SvelteSet());

// Somewhere further down
myOptionalSet = null; // This needs the variable to be wrapped in a $state in order for this to be reactive.
```

**Comment by mikededo** [comment]
I agree with @marekdedic. The statement that reactive built-ins _do not need to be wrapped_ with `$state` is not correct.

**Comment by marekdedic** [comment]
Yes, at the same time, it actually is a mistake like _95% of the time_... So I think it's a great candidate for a lint rule, the rule just needs to be clever and actually figure out when the `$state` is really not needed.
